src/ListValidator.php

```diff
@@ -25,13 +29,27 @@ public function doValidation( $value ) {
 			return;
 		}
 
-		$optionMap = array(
-			'elementcount' => 'range',
-			'maxelements' => 'upperbound',
-			'minelements' => 'lowerbound',
-		);
+		$validator = new RangeValidator();
+
+		if ( array_key_exists( 'elementcount', $this->options ) ) {
+			$range = $this->options['elementcount'];
+
+			if ( !is_array( $range ) || count( $range ) !== 2 ) {
+				throw new Exception( 'The elementcount option must be an array with two elements' );
```

Comment:
I wanted to have this patch 100% compatible to the original implementation and copy-pasted the line https://github.com/DataValues/Validators/blob/master/src/RangeValidator.php#L146, which was triggered before. That's why there is `$this->setExpectedException( 'Exception' )` in the test.

But you are right, I can change this to `InvalidArgumentException`. This will not be a breaking change, right?